pkg/prometheus/server/operator.go

```diff
@@ -972,6 +986,45 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+func (c *Operator) shouldDelete(p *monitoringv1.Prometheus, sts *appsv1.StatefulSet) (bool, error) {
+	if !c.retentionPoliciesEnabled {
+		// Feature-gate is disabled, default behavior is always to delete.
+		return true, nil
+	}
+
+	srp := p.Spec.ShardRetentionPolicy
+	if srp == nil || srp.WhenScaledDown == monitoringv1.WhenScaledDownRetentionTypeDelete {
+		return true, nil
+	}
+
+	retentionAnnotation, ok := sts.Annotations[deletionTimestampAnnotation]
+	if !ok {
+		// This statefulset was not marked for deletion yet. Let's just add the annotation and return.
+		sts.Annotations[deletionTimestampAnnotation] = c.getRetentionDate(p)
```

Comment:
the statefulset is never updated though? Should this PR only deal with retain vs. delete and we address the retention timestamp in a follow-up?

Reply:
Yeah good point, I think I misjudged how this function was used when I was making the scope of this PR smaller.

I'll delete this and re-add once we deal with timestamps